tests: Add test for drivers/mm/mm_drv_bank module

peter-mitsis opened the pull request:
Adds a test for the drivers/mm/mm_drv_bank module.

peter-mitsis committed:
drivers: mm: Fix include header guard

Fixes the include header guard so that it does not use
kernel.h's.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

peter-mitsis committed:
tests: Add test for mm_drv_bank

Adds a test for verifying that the mm_drv_bank APIs properly
record data for mapped and unmapped pages.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

galak commented on tests/drivers/mm/sys_mm_drv_bank/testcase.yaml:
```diff
@@ -0,0 +1,3 @@
+tests:
+  drivers.mm.sys_mm_drv_bank:
```
Seems like we should have some filter so we only build / run this on platforms that have drivers/mm.

galak replied:
with arm-clang we get errors of the form:
```
Error: L6218E: Undefined symbol sys_mm_drv_page_phys_get (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_map_page (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_unmap_page (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_page_flag_get (referred from mm_drv_common.o).
Error: L6218E: Undefined symbol sys_mm_drv_update_page_flags (referred from mm_drv_common.o).
```

galak reviewed (changes requested):
Need to filter this for platforms that actually have a "driver" that exists.  Otherwise we get build failures on arm-clang.

galak commented on tests/drivers/mm/sys_mm_drv_bank/testcase.yaml:
```diff
@@ -0,0 +1,4 @@
+tests:
+  drivers.mm.sys_mm_drv_bank:
+    tags: drivers mm
+    filter: CONFIG_MM_DRV
```
`filter: CONFIG_MM_DRV` does not solve the issue.   I think adding something like:

```
filter: dt_compat_enabled("intel,adsp-tlb") or dt_compat_enabled("intel,adsp-mtl-tlb")
```

We need to add this to `tests/drivers/mm/sys_mm_drv_api/testcase.yaml`

peter-mitsis replied:
Gotcha. Hopefully this will do it. :)

peter-mitsis committed:
tests: Add filter to sys_mm_drv_api test

Adds a filter to the sys_mm_drv_api test to restrict it
it intel adsp mtl platforms.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

galak commented on tests/drivers/mm/sys_mm_drv_api/testcase.yaml:
```diff
@@ -1,3 +1,4 @@
 tests:
   drivers.mm.sys_mm_drv_api:
     tags: drivers mm
+    filter: dt_compat_enabled("intel,adsp-mtl-tlb")
```
wny not `or dt_compat_enabled("intel,adsp-tlb")`

galak commented on tests/drivers/mm/sys_mm_drv_bank/testcase.yaml:
```diff
@@ -0,0 +1,4 @@
+tests:
+  drivers.mm.sys_mm_drv_bank:
+    tags: drivers mm
+    filter: dt_compat_enabled("intel,adsp-mtl-tlb")
```
wny not or dt_compat_enabled("intel,adsp-tlb")
